crates/storage/codecs/derive/src/compact/structs.rs

```diff
@@ -116,17 +116,26 @@ impl<'a> StructHandler<'a> {
             format_ident!("specialized_from_compact")
         };
 
+        // ! Be careful before changing the following assert ! Especially if the type does not
```

Comment:
love it - seems fine to have footgun at codec-related logic, is low level enough and i like that we're highlighting it